### PR TITLE
Keeps popup open if it has keyboard focus

### PR DESCRIPTION
**Pull request by WyohKnott**
Fixes https://github.com/new-xkit/XKit/issues/122

### CODE REVIEW

File: Extensions/quick_tags.js [hunk 1]
```diff
@@ -297,9 +302,6 @@ XKit.extensions.quick_tags = new Object({
 		$("#xkit-quick-tags-window").remove();
 		$("body").append(m_html);
 
-		$("#xkit-tag-input").bind("focus", XKit.extensions.quick_tags.cancel_menu_close);
-		$("#xkit-tag-input").bind("blur", XKit.extensions.quick_tags.menu_close);
```
Comment (hobinjk) [inline Extensions/quick_tags.js]: I understand the removal of the focus event, but doesn't removing the blur event mean that clicking outside of the quick tags menu to blur the input will no longer close the quick tags menu?

File: Extensions/quick_tags.js [hunk 2]
```diff
@@ -47,7 +47,12 @@ XKit.extensions.quick_tags = new Object({
 	},
 
 	menu_close: function() {
-		XKit.extensions.quick_tags.menu_closer_int = setTimeout(function() { XKit.extensions.quick_tags.close_window(); }, 500);
+		// Only close the menu if it doesn't have keyboard or mouse focus
+		if ($("#xkit-quick-tags-window").find('input:focus').length === 0 &&
+		    $('#xkit-quick-tags-window:hover').length === 0) {
+			XKit.extensions.quick_tags.user_on_box = false;
```
Comment (hobinjk) [inline Extensions/quick_tags.js]: Nitpick: the indentation doesn't match up between this and the following line

File: Extensions/quick_tags.js (hunk 1 above)
Comment (WyohKnott) [inline Extensions/quick_tags.js]: That's the goal. The behaviour is now similar to one click postage, i.e. if one click outside of the popup, it closes, *unless* the focus was given to the input area, in which case it won't close (in order to do some copy-paste for example)

**Comment by nightpool** [comment]
Does the quick tags menu ever get keyboard focus though? I tried tagging some stuff with the new extension and I couldn't notice a difference.

File: Extensions/quick_tags.js (hunk 1 above)
Comment (nightpool) [inline Extensions/quick_tags.js]: nevermind i get it now.

**Comment by hobinjk** [comment]
Great, merging.

**Comment by Federer343** [comment]
Is there any way to make this so you can turn it off?  I like how the old way did it much more, having the pop-up box remain up after hitting enter if really annoying.

**Pull request merged**